core/server/src/main/java/alluxio/master/file/meta/InodeDirectory.java

```diff
@@ -76,15 +77,20 @@ protected InodeDirectory getThis() {
    * @param child the inode to add
    */
   public void addChild(Inode<?> child) {
-    mChildren.add(child);
+    synchronized (child) {
```

Comment:
We can hide this synchronize if we let the index add returns false when facing a duplicated value instead of throwing an exception. Should we do that?